index.js

```diff
@@ -10,7 +10,7 @@ var has = Object.prototype.hasOwnProperty;
  * @api public
  */
 function querystring(query) {
-  var parser = /([^=?&]+)=([^&]*)/g
+  var parser = /([^=?&]+)=?([^&]*)?/g
```

Comment:
I think we can remove the final `?`. The second capturing group will be the empty string when there is no value.